sui/src/benchmark.rs

```diff
@@ -0,0 +1,289 @@
+    }
+}
+
+fn run_latency_microbench(
+    network_client: NetworkClient,
+    network_server: NetworkServer,
+    connections: usize,
+    use_move: bool,
+    committee_size: usize,
+    db_cpus: usize,
+
+    num_chunks: usize,
+    chunk_size: usize,
+    period_us: u64,
+) -> MicroBenchmarkResult {
+    assert!(
+        (num_chunks * chunk_size % connections) == 0,
+        "num_transactions must {} be multiple of number of TCP connections {}",
+        num_chunks * chunk_size,
+        connections
+    );
+    let mut tx_cr = TransactionCreator::new(committee_size, db_cpus);
+
+    // These TXes are to load the network
+    let load_gen_txes = tx_cr.generate_transactions(connections, use_move, chunk_size, num_chunks);
+
+    // These are tracer TXes used for measuring latency
+    let tracer_txes = tx_cr.generate_transactions(1, use_move, 1, num_chunks);
+
+    // Make multi-threaded runtime for the authority
+    thread::spawn(move || {
+        let runtime = Builder::new_multi_thread()
+            .enable_all()
+            .thread_stack_size(32 * 1024 * 1024)
+            .build()
+            .unwrap();
+
+        runtime.block_on(async move {
+            let server = spawn_authority_server(network_server, tx_cr.authority_state).await;
+            if let Err(e) = server.join().await {
+                error!("Server ended with an error: {e}");
+            }
+        });
+    });
+
+    // Wait for server start
+    sleep(Duration::from_secs(3));
+
+    let runtime = Builder::new_multi_thread()
+        .enable_all()
+        .thread_stack_size(32 * 1024 * 1024)
+        .worker_threads(usize::min(num_cpus::get(), 24))
+        .build()
+        .unwrap();
+
+    // Prep the generators
+    let (mut load_gen, mut tracer_gen) = runtime.block_on(async move {
+        join!(
+            FixedRateLoadGenerator::new(
+                load_gen_txes,
+                period_us,
+                network_client.clone(),
+                connections,
+            ),
+            FixedRateLoadGenerator::new(tracer_txes, period_us, network_client, 1),
+        )
+    });
+
+    // Run the load gen and tracers
+    let (load_latencies, tracer_latencies) =
+        runtime.block_on(async move { join!(load_gen.start(), tracer_gen.start()) });
+
+    MicroBenchmarkResult::Latency {
+        load_chunk_size: chunk_size,
+        load_latencies,
+        tick_period_us: period_us as usize,
+        chunk_latencies: tracer_latencies,
+    }
+}
+
+async fn run_follower(network_client: NetworkClient) {
+    // We spawn a second client that listens to the batch interface
+    let _batch_client_handle = tokio::task::spawn(async move {
+        let authority_client = AuthorityClient::new(network_client);
+
+        let mut start = 0;
+
+        loop {
+            let receiver = authority_client
+                .handle_batch_streaming_as_stream(BatchInfoRequest {
+                    start,
+                    end: start + 10_000,
```

Comment:
What is the 10_000?

Reply:
This was added by @gdanezis 
I simply ported it over.
In reality we want to benchmark this follower logic more thoroughly but thats still in the works